components/brave_wallet_ui/page/screens/onboarding/restore-from-recovery-phrase/restore-from-recovery-phrase.tsx

```diff
@@ -0,0 +1,508 @@
+                </PhraseCardTopRow>
+
+                <PhraseCardBody>
+                  {RecoveryInput}
+                </PhraseCardBody>
+
+                <PhraseCardBottomRow centered>
+                  <WalletLink
+                    as='button'
+                    onClick={onClickPasteFromClipboard}
+                  >
+                    {getLocale('braveWalletPasteFromClipboard')}
+                  </WalletLink>
+                </PhraseCardBottomRow>
+              </PhraseCard>
+
+              <VerticalSpace space={isPhraseShown ? '20px' : '130px'} />
+            </>
+          }
+
+          {!isCheckingExtensions && currentStep === RestoreFromOtherWalletSteps.currentPassword &&
+            <>
+              <PasswordInput
+                autoFocus={true}
+                onChange={onExtensionPasswordChange}
+                value={extensionPassword}
+                error={importWalletError?.errorMessage || ''}
+                hasError={importWalletError?.hasError}
+                onKeyDown={handleKeyDown}
+                placeholder={
+                  restoreFrom === 'metamask'
+                    ? getLocale('braveWalletMetaMaskPasswordInputPlaceholder')
+                    : getLocale('braveWalletImportBraveLegacyInput')
+                }
+                name='extensionPassword'
+                label={getLocale('braveWalletInputLabelPassword')}
+              />
+
+              <VerticalSpace space='100px' />
+            </>
+          }
+
+          {currentStep === RestoreFromOtherWalletSteps.newPassword &&
+            <>
+              <NewPasswordInput
+                autoFocus={true}
+                onSubmit={restoreWallet}
+                onChange={handlePasswordChange}
+              />
+              {importWalletError?.hasError &&
+                <ErrorText>
+                  {importWalletError.errorMessage}
+                </ErrorText>
+              }
+            </>
+          }
+
+          {!isCheckingExtensions &&
+            <NextButtonRow>
+              <NavButton
+                buttonType='primary'
+                text={isCheckingImportPassword
+                  ? <LoadingIcon
+                    size='24px'
+                    opacity={0.8}
+                    color='interactive08'
+                  /> as unknown as string
+                  : getLocale('braveWalletButtonContinue')
+                }
+                onSubmit={onContinueClicked}
+                disabled={
+                  isCheckingImportPassword ||
+                  isImportingFromExtension && currentStep === RestoreFromOtherWalletSteps.currentPassword && !extensionPassword ||
+                  restoreFrom === 'seed' && !phraseInput ||
+                  currentStep === RestoreFromOtherWalletSteps.phrase && invalidMnemonic ||
+                  currentStep === RestoreFromOtherWalletSteps.newPassword && !isPasswordValid
+                }
+              />
+            </NextButtonRow>
+          }
```

Comment:
New wallet restore implementation. Will now check the import password before allowing the creation of a new password

Reply:
@kdenhartog  recovery phrase used